File: scala/tst/RandomGraphGeneratorTest.scala
class RandomGraphGeneratorTest extends org.scalatest.FunSuite {
  test("random_graph") {
    RandomGraphGenerator.new_undirected(2, 0)
    RandomGraphGenerator.new_undirected(3, 0)
    RandomGraphGenerator.new_undirected(3, 1)
    RandomGraphGenerator.new_undirected(3, 2)
    RandomGraphGenerator.new_undirected(4, 0)
    RandomGraphGenerator.new_undirected(4, 1)
    RandomGraphGenerator.new_undirected(4, 2)
    RandomGraphGenerator.new_undirected(4, 3)
    RandomGraphGenerator.new_undirected(4, 4)
    RandomGraphGenerator.new_undirected(4, 5)
  }
}

File: java/src/test/java/be/steinsomers/bron_kerbosch/study/RandomGraphGeneratorTest.java
package be.steinsomers.bron_kerbosch.study;

import org.junit.jupiter.api.Test;

import java.util.Random;

final class RandomGraphGeneratorTest {
    @Test
    void random_undirected_graph() {
        var rng = new Random(19680516);
        var gen = new RandomGraphGenerator(rng);
        gen.newUndirected(3, 0);
        gen.newUndirected(3, 1);
        gen.newUndirected(3, 2);
        gen.newUndirected(4, 0);
        gen.newUndirected(4, 1);
        gen.newUndirected(4, 2);
        gen.newUndirected(4, 3);
        gen.newUndirected(4, 4);
        gen.newUndirected(4, 5);
    }
}
